ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
+        }
+
         $button = array(
           array(
             '#theme' => 'link',
             '#text' => t('Reserve'),
-            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $path),
+            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . base64_encode(implode('|', $item_id)),
```

Comment:
Assuming [we remove rawurlencode from `ding_periodical`](https://github.com/ding2/ding_periodical/pull/6/files#r40773655) then I see the situation as follows:

1. Drupal does not handle `/` in URLs.
2. To avoid this we take the 3 elements which make up a periodical id, `implode` and `base64_encode` them into a single value
3. We decode and `explode` the value when processing requests in `ding_periodical_reserve_ajax()` to continue working with the three values.

My comments:

1. Reducing the number of page arguments from 3 to 1 makes the code less robust. We could code our own checks to ensure we have the expected number of values but the pull request does not contain this at the moment. I think a better solution would be to encode each af the three arguments separately.
2. I do not think `base64_encode()` is the best approach to encoding these values. Each argument becomes longer and less legible (relevant when debugging). How about using `urlencode()` or `rawurlencode()` and the appropriate decode functionin `ding_periodical_reserve_ajax`? As I see it `/` would still be encoded, thus fixing the bug and issues regarding `base64_encode()` would be avoided.